src/rebootstrap-raft/getstatepassword_test.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"errors"
+	"io"
+	"strings"
+	"testing"
+)
+
+func TestGetStatePasswordMissingFile(t *testing.T) {
+	openFile = func(string) (io.ReadCloser, error) {
+		return nil, errors.New("file not found")
+	}
+	command := &rebootstrapCommand{}
+	_, err := command.getStatePassword("0")
+	if err == nil {
+		t.Errorf("Expected 'file not found' error")
+	}
+	if err.Error() != "opening agent config file \"/var/lib/juju/agents/machine-0/agent.conf\": file not found" {
+		t.Errorf("Expected error, got %v", err)
+	}
+}
+
+func TestGetStatePassword(t *testing.T) {
+	openFile = func(string) (io.ReadCloser, error) {
+		content := `# format 2.0
+tag: machine-0
+datadir: /var/lib/juju
+transient-datadir: /var/run/juju
+logdir: /var/log/juju
+metricsspooldir: /var/lib/juju/metricspool
+nonce: user-admin:bootstrap
+jobs:
+- JobManageModel
+- JobHostUnits
+upgradedToVersion: 2.9.51
+statepassword: qKoU8Y/BRQd0shDFA4OdHxlf
+controller: controller-e5e3d855-9df1-4e24-8010-0bfa6bc3cbf1
+model: model-a03dcd4f-afaf-4c96-8965-f2537e3b4873
+apiaddresses:
+- 10.1.20.117:17070
+- '[fd42::216:3eff:fe69:bb55]:17070'
+apipassword: qKoU8Y/BRQd0shDFA4OdHxlf
+oldpassword: 8e2e2bdb1e690dcee14c381bbe92baf5
+loggingconfig: <root>=INFO; unit=DEBUG
+values:
+  AGENT_SERVICE_NAME: jujud-machine-0
+  CONTAINER_TYPE: ""
+  NUMA_CTL_PREFERENCE: "false"
+  PROVIDER_TYPE: lxd
+agent-logfile-max-size: 100
+agent-logfile-max-backups: 2
+apiport: 17070
+stateport: 37017
+mongoversion: 4.4.24/wiredTiger
+mongomemoryprofile: default
+juju-db-snap-channel: 4.4/stable`
+		return io.NopCloser(strings.NewReader(content)), nil
+	}
+	command := &rebootstrapCommand{}
+	password, err := command.getStatePassword("0")
```

Comment:
My thought here was that instead of testing getStatePassword(machineId), you would test a different "extractStatePassword(io.Reader()"
and then we could pass our own content directly into it.
We could still leave getStatePassword() as something that calls extract but then you can test the parser without changing how the system reads and rights to the filesystem.